### ISSUE TITLE
Why there are so many memory leaks....

### ISSUE THREAD

**Issue by SimonNie98** [CMU-SAFARI/MQSim#59]
Jesus, when i try to run the porgram with valgrind or address sanitizer, there are so many memory leaks. And there are so many "new" without "delete"...

**Comment by JJJKJKJ** [comment]
Yes, I have encountered the same problem. In a workload configuration file, only 20 concurrent four loads were placed, causing memory accumulation to skyrocket. I'm not sure if you have solved this problem. If so, can I get your solution
